### PR TITLE
IntegrationSpec and N+1

### PR DESCRIPTION
**Pull request by samlorlah**
### In this pull request, We have implemented the changes: 
Screenshot 2023-02-08 at 8 22 25 AM

Screenshot 2023-02-08 at 8 22 37 AM

- Add integration testing for each view in my project
   - Add test for text and image content: 
      - Add test to see all `usernames `of users in(user index, user show, post index and post show) 
      - Add test to see all `profile pictures` for each user 
      - Add test to see `number` of posts each user has written
  - Add test for clicks and redirect: 
     - Add test in user index page to get redirected to user’s show page when a user is clicked
     - Add test in user show page to get redirected to a post’s show page when a user’s post is clicked 
     - Add test in user show page to get redirected to a user’s post index page when the `see all post` button is clicked 
     - Add test in post index page to get redirected to a post’s show page when a specific post is clicked
  - Add test to counts 
     - Add test for like_counter, comment_counter and post_counter

### CODE REVIEW

**Review by prolajumokeoni — COMMENTED** [review]
Hi @Team

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request
You have submitted a project, that does not have the  key features section of your app in your readme

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/views/posts/user_post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    end
+    it 'should show how many comments a post has' do
+      post = @user.posts.first
+      expect(page).to have_content("Comments: #{post.comments_counter}")
+    end
+    it 'should have button for pagination' do
+      expect(page).to have_button('Pagination')
+    end
+  end
+end
```
Comment (Nemwel-Boniface) [inline spec/views/posts/user_post_index_spec.rb]: - [x] You might have forgotten to add the tests for "I can see how many likes a post has" Kindly include it.

image

- [x] You might have forgotten to add the tests for "When I click on a post, it redirects me to that post's show page." Kindly include it.

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @samlorlah  and @chuaindia       :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Good job working on your project so far! :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

- [ ] For some reason, when I run ```rspec spec/views``` to check if the tests are running correctly they are failing. Below is the screenshot of the result I am getting. Kindly look into why this is happening.

image


_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/posts/user_post_index_spec.rb (hunk 1 above)
Comment (samlorlah) [reply spec/views/posts/user_post_index_spec.rb]: I have added the 2 tests.

Please note that all tests are passing here. 

Screenshot 2023-02-08 at 12 35 19 PM

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @samlorlah👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- All tests are passed. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**